x-pack/filebeat/input/httpjson/input_test.go

```diff
@@ -694,6 +694,91 @@ func TestInput(t *testing.T) {
 				`{"space":{"cake":"pumpkin"}}`,
 			},
 		},
+		{
+			name: "Test global transform context separation with parent_last_response object",
+			setupServer: func(t *testing.T, h http.HandlerFunc, config map[string]interface{}) {
+				var serverURL string
+				registerPaginationTransforms()
+				registerRequestTransforms()
+				r := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					switch r.URL.Path {
+					case "/":
+						link := serverURL + "/link1"
+						value := fmt.Sprintf(`{"files":[{"id":"1"},{"id":"2"}],"exportId":"2212", "nextLink":"%s"}`, link)
+						fmt.Fprintln(w, value)
+					case "/link1":
+						fmt.Fprintln(w, `{"files":[{"id":"3"},{"id":"4"}], "exportId":"2213"}`)
+					case "/2212/1":
+						body, _ := io.ReadAll(r.Body)
+						r.Body.Close()
+						if string(body) == `{"exportId":"2212"}` {
+							fmt.Fprintln(w, `{"hello":{"world":"moon"}}`)
+						}
```

Comment:
A little helper here would improve readability.
```
func matchBody(w io.Writer, req *http.Request, match, response string) {
    body, _ := io.ReadAll(req.Body)
    req.Body.Close(0
    if string(body) == match {
        w.Write([]byte(response))
    }
}
```

This would allow these mux patterns to be handled like this
```
case "/2212/1":
    matchBody(w, r, `{"exportId":"2212"}`, `{"hello":{"world":"moon"}}`)
```
and so on. This way we get the explicit behaviour description, but don't have to deal with the messy details in the test case.